src/desktop/dialogs/versioncheckdialog.cpp

```diff
@@ -53,11 +39,27 @@ VersionCheckDialog::~VersionCheckDialog()
 
 void VersionCheckDialog::rememberSettings()
 {
-	QSettings().setValue("versioncheck/enabled", !m_ui->dontCheck->isChecked());
+	QSettings().setValue("versioncheck/enabled", m_ui->checkForUpdates->isChecked());
 }
 
 void VersionCheckDialog::doVersionCheckIfNeeded()
 {
+	if(!QSettings().contains("versioncheck/enabled")) {
```

Comment:
An alternative option which I prefer would be to do something like enhancing the notification bar and putting the question there, since modals are needlessly confrontational and the software can run without the user making this decision, so it should.

I’m not a big fan of first run wizards so would like to get more information about the translation issue. Right now the locale detection definitely isn’t as robust as it theoretically should be and having a concrete example would be helpful.

With regards to the “can’t join friends”, IIRC there is/was a deficiency in the current UI where there is no clear CTA about the version mismatch unless (1) auto-update is turned on and (2) it saw there was a new update and (3) someone decided to not update. So solving this particular problem, probably best to show a notice whenever a newer protocol is seen, either by just showing it all the time when a newer protocol is seen, or by making newer protocol sessions selectable and then disabling the join button and displaying a CTA when someone selects a too-new session so it’s obvious what is going on (right now, disabling with a weird question mark icon that has a tooltip is bad UX).

Reply:
Putting it in an unobtrusive place also seems fine to me, having it yell at you when it notices a newer version also makes a lot of sense.

Translations are community-provided, so I think there was something wrong with Portuguese somehow being mixed with French or something, and I don't think we distinguish things properly, like only having one Chinese translation when there sure is plenty more than one Chinese language. I was thinking about looking into Weblate to let people more easily contribute translations, but that's another story.

Reply:
I have added a TODO to make the opt-in less obtrusive soon.